Assets/Scripts/Extentions/VectorExtensions.cs

```diff
@@ -309,13 +332,12 @@ public static bool CompareZ(this Vector3 v, Vector3 f, float margin = 0.0001f)
         /// <param name="yRange">Range for the Y axis.</param>
         /// <param name="zRange">Range for the Z axis.</param>
         /// <returns>A new Vector3 with randomized axes.</returns>
-        public static Vector3 RandomizeVector(this Vector3 v, Vector2 xRange, Vector2 yRange, Vector2 zRange)
+        public static ref Vector3 RandomizeVector(ref this Vector3 v, Vector2 xRange, Vector2 yRange, Vector2 zRange)
         {
-            return new Vector3(
-                Random.Range(xRange.x, xRange.y),
-                Random.Range(yRange.x, yRange.y),
-                Random.Range(zRange.x, zRange.y)
-            );
+            v.x = Random.Range(xRange.x, xRange.y);
+            v.y = Random.Range(yRange.x, yRange.y);
+            v.z = Random.Range(zRange.x, zRange.y);
```

Comment:
Why doesn't this for example make use of the RandomizeX,Y,Z methods you already made?